### ISSUE TITLE
Cannot save multiple versionned records types with same primary key under different subspace prefix in a same transaction

### ISSUE THREAD

**Issue by sebastienh** [FoundationDB/fdb-record-layer#1351]
# Cannot save multiple versionned records types with same primary key under different subspace prefix in a same transaction

Note: _This issue has been found in the last maven published foundationdb record layer version (2.8.110.0). If it has been fixed since then, please disregard this issue. I will upgrade when the next version becomes available._

I have an fdb transaction where I try to save mutliple records of different types. Each record type is stored under a different record store under a different keypath. These records all have the same primary key but since they are stored under a different keypath they are all locally unique inside their own record store. The record layer does not accept to save and give the error "Found split record without start" in SplitHelper.java. It seems that because these records share a same primary key, the SplitHelper thinks that they come from a split record and does not accept to save them. In fact, I try to use local version, and since all my records all have the same "primary key" `context.getLocalVersion(key)` returns the same local version for all the records. (I use `context.claimLocalVersion()` to make sure each localVersion is unique inside a transaction)

In `SplitHelper.java`:

```java
    context.getLocalVersion(key).ifPresent(localVersion -> {
        version = FDBRecordVersion.incomplete(localVersion);
        sizeInfo.setVersionedInline(true);
        sizeInfo.keyCount += 1;
        sizeInfo.keySize += keySplitSubspace.pack(RECORD_VERSION).length;
        sizeInfo.valueSize += 1 + FDBRecordVersion.VERSION_LENGTH;
    });
```

It seems to me that the Record Layer should not make any assumptions about how the primary keys are defined inside each record store and use the subspace to distinguish between the different local version in the current context (FDBRecordContext).  

Thanks for your help!

**Comment by MMcM** [comment]
This sounds similar to #782, which was fixed in 2.8.94.0.

**Comment by MMcM** [comment]
Sounds similar to #952, which was indeed fixed in `2.10.135.0`.

**Comment by sebastienh** [comment]
Yes, it's the same issue! Ok, so I will wait then.  Thanks!
